src/metabase/models/dashboard.clj

```diff
@@ -285,24 +285,33 @@
         (update-field-values-for-on-demand-dbs! old-param-field-ids new-param-field-ids)))))
 
 (defn update-dashcards!
```

Comment:
IMO long `let` form makes it hard to read because it makes the reader have to dereference and keep track of multiple variables. 

And also some of these bindings are only used once so I think we could shorten the let form a bit like this

```clojure
(defn update-dashcards!
  "Update the `dashcards` belonging to `dashboard`.
  This function is provided as a convenience instead of doing this yourself; it also makes sure various cleanup steps
  are performed when finished, for example updating FieldValues for On-Demand DBs.
  Returns `nil`."
  {:style/indent 1}
  [dashboard new-dashcards]
  (let [dashboard                  (t2/hydrate dashboard [:ordered_cards :series :card])
        old-dashcards              (:ordered_cards dashboard)
        id->old-dashcard           (m/index-by :id old-dashcards)]
    ;; ensure the dashcards we are updating are part of the given dashboard
    (when (set/subset? (set (map :id new-dashcards))
                       (set (keys id->old-dashcard)))
      (throw (ex-info (tru "Dashboard {0} does not have a DashboardCard with ID {1}"
                           (u/the-id dashboard) (first only-new))
                      {:status-code 404})))
    (doseq [dashcard new-dashcards]
      (let [;; update-dashboard-card! requires series to be a sequence of card IDs
            old-dashcard       (-> (get id->old-dashcard (:id dashcard))
                                   (update :series #(map :id %)))
            dashboard-card     (update dashcard :series #(map :id %))]
        (dashboard-card/update-dashboard-card! dashboard-card old-dashcard)))
    (let [new-param-field-ids (params/dashcards->param-field-ids (t2/hydrate new-dashcards :card))]
      (update-field-values-for-on-demand-dbs! (params/dashcards->param-field-ids old-dashcards) new-param-field-ids))))
```